### PR TITLE
Download drug consumption report as CSV

### PR DESCRIPTION
**Pull request by prabhanshuguptagit**
**Story card:** [ch3624](https://app.clubhouse.io/simpledotorg/story/3624/allow-downloading-drug-consumption-report)

## Because

Drug consumption reports to be downloaded as a CSV.

## This addresses

Adds: `/my_facilities/drug_consumption/download.csv`

### CODE REVIEW

**Review by harimohanraj89 — CHANGES_REQUESTED** [review]
Some minor feedback. The CSV generation is a bit cumbersome, but it's very clear what's going on. I took a stab at massaging the `@drugs_by_category` into a data structure that would be more easily cast to CSV, but the dynamic nature of categories and drugs makes it a little hard.

File: app/exporters/drug_consumption_report_exporter.rb [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    ["Facilities"] + drug_name_cells + drug_category_name_cells
+  end
+
+  def total_consumption_row
+    drug_consumption_cells =
+      @drugs_by_category.flat_map do |drug_category, drugs|
+        drugs.map do |drug|
+          consumed = @report[:all].dig(drug_category, drug, :consumed)
+          next "?" if consumed.nil? || consumed == "error"
+          consumed
```
[the remark this answers is not in the document]
Comment (prabhanshuguptagit) [reply app/exporters/drug_consumption_report_exporter.rb]: The HTML view does some fancy styling on top, extracted this to a method in the exporter.

File: app/controllers/my_facilities/drug_stocks_controller.rb [hunk 2]
```diff
@@ -31,6 +31,16 @@ def download_drug_stock
...
 
+  def download_drug_consumption
+    create_drug_report
+
+    respond_to do |format|
+      format.csv do
+        send_data DrugConsumptionReportExporter.csv(@query), filename: "drug-consumption-report-#{@for_end_of_month_display}.csv"
+      end
+    end
+  end
```
[the remark this answers is not in the document]
Comment (prabhanshuguptagit) [reply app/controllers/my_facilities/drug_stocks_controller.rb:36]: Added in 85bd94b

**Review by harimohanraj89 — APPROVED** [review]
Godspeed

**Pull request merged**